src/big/big_decimal.cr

```diff
@@ -47,34 +47,79 @@ struct BigDecimal < Number
   def initialize(str : String)
     raise InvalidBigDecimalException.new(str, "Zero size") if str.bytesize == 0
 
-    # Check str's validity and find index of .
+    # Check str's validity and find index of '.'
     decimal_index = nil
+    # Check str's validity and find index of 'e'
+    exponent_index = nil
+
     str.each_char_with_index do |char, index|
       case char
       when '-'
-        if index != 0
+        unless index == 0 || exponent_index.try &.==(index - 1)
           raise InvalidBigDecimalException.new(str, "Unexpected '-' character")
         end
+      when '+'
+        unless exponent_index.try &.==(index - 1)
+          raise InvalidBigDecimalException.new(str, "Unexpected '+' character")
+        end
       when '.'
         if decimal_index
           raise InvalidBigDecimalException.new(str, "Unexpected '.' character")
         end
-
         decimal_index = index
+      when 'e'
```

Comment:
I think capital E is usually supported in these notations, so we should add it (and add a spec)

Reply:
Done.